examples/workbooks/LeadsExample.ts

```diff
@@ -0,0 +1,325 @@
+            fns.map((f) => f(x));
+
+/**
+ * Converts the normal `String.prototype.toLowerCase()` to a normal fn so it can be used with `pipe`.
+ *
+ * @param {string} value - value to apply operation.
+ *
+ * @example
+ * pipe(value, toLowerCase);
+ */
+const toLowerCase = (value: string): string => value.toLowerCase();
+
+/**
+ * Converts the normal `String.prototype.trim()` to a normal fn so it can be used with `pipe`.
+ *
+ * @param {string} value - value to apply operation.
+ *
+ * @example
+ * pipe(value, trim);
+ */
+const trim = (value: string): string => value.trim();
+
+/**
+ * Allows us to combine multiple validations in a quick and easy way.
+ *
+ * @example
+ * runValidations(fn1, fn2, fn3, ...);
+ */
+const runValidations = (...fns: Array<any>): Array<FF.Message> => {
+    return fns.reduce((acc, fn) => [...acc, fn()], []).filter(isNotNil);
+};
+
+/*
+ * Main
+ */
+
+const LeadsSheet = new FF.Sheet(
+    "Leads",
+    {
+        first_name: FF.TextField({
+            label: "First Name",
+            description: "Lead's first name.",
+        }),
+        last_name: FF.TextField({
+            label: "Last Name",
+            description: "Lead's last name.",
+            required: true,
+        }),
+        email: FF.TextField({
+            label: "Email Address",
+            description: "Lead's email.",
+            unique: true,
+            compute: (value) => {
+                return pipe(value, trim, toLowerCase);
+            },
+            validate: (value) => {
+                const ensureValidEmail = validateEmail(value);
+
+                return runValidations(ensureValidEmail);
+            },
+        }),
+        phone: FF.TextField({
+            label: "Phone Number",
+            description: "Lead's phone.",
+        }),
+        date: FF.DateField({
+            label: "Date",
+            description: "Date goes here.",
+        }),
+        country: FF.TextField({
+            label: "Country",
+            description: "Country goes here",
+            cast: FF.CountryCast("iso-2"),
+        }),
+        postal_code: FF.TextField({
+            label: "Postal Code",
+            description: "Postal code goes here",
+        }),
+        opt_in: FF.BooleanField({
+            label: "Opt In",
+            description: "Opt in goes here",
+        }),
+        deal_status: FF.OptionField({
+            label: "Deal Status",
+            description: "Deal status goes here",
+            options: {
+                prospecting: "Prospecting",
+                discovery: "Discovery",
+                proposal: "Proposal",
+                negotiation: "Negotiation",
+                closed_won: "Closed Won",
+                closed_lost: "Closed Lost",
+            },
+        }),
+    },
+    {
+        allowCustomFields: false,
+        readOnly: true,
+        recordCompute: (record, _session, _logger) => {
+            return fold(emailOrPhoneRequired, zipCodeZeroPadding)(record);
```

Comment:
What do you think the recommended recordHooksCombine should look like, I would write it with that name.

Fold or pipe are likely to be confusing to users who aren't familiar with functional programming.

a function that runs recordComputes in order, and try/catch protects them could be useful
it could also be useful to stop processing onError.  those two combination methods could be combined (they each take a list of recordCompute functions and return a single recordCompute function) by the user.

Reply:
There are a lot of ways to do a combination / concatenation. Let's not lose the forest for the trees here. If this comes with a description of what it is doing and solves the problem reliably, I would avoid trying to find the "best" possible way to do this, and save that debate for a method that we publish to the SDK. I've seen some prospects literally just stringify everything and use + to put two fields together 😂